packages/transformer/src/DetachedExportElementAspectsStrategy.ts

```diff
@@ -0,0 +1,108 @@
+  onExportElementUniqueAspect(uniqueAspect: ElementUniqueAspect, isUpdate?: boolean | undefined): void;
+  onExportElementMultiAspects(multiAspects: ElementMultiAspect[]): void;
+  trackProgress(): Promise<void>;
+}
+
+/**
+ * Detached ElementAspect export strategy for [[IModelExporter]].
+ * This strategy exports all ElementAspects separately from the Elements that thei are connected with.
+ * @internal
+ */
+export class DetachedExportElementAspectsStrategy extends ExportElementAspectsStrategy<DetachedElementAspectsHandler> {
+  public async exportAllElementAspects(): Promise<void> {
+    await this.exportAspectsLoop<ElementUniqueAspect>(ElementUniqueAspect.classFullName, async (uniqueAspect) => {
+      const isInsertChange = this.aspectChanges?.insertIds.has(uniqueAspect.id) ?? false;
+      const isUpdateChange = this.aspectChanges?.updateIds.has(uniqueAspect.id) ?? false;
+      const doExport = this.aspectChanges === undefined || isInsertChange || isUpdateChange;
+      if (doExport) {
+        const isKnownUpdate = this.aspectChanges ? isUpdateChange : undefined;
+        this.handler.onExportElementUniqueAspect(uniqueAspect, isKnownUpdate);
+        await this.handler.trackProgress();
+      }
+    });
+
+    let batchedElementMultiAspects: ElementMultiAspect[] = [];
+    await this.exportAspectsLoop<ElementMultiAspect>(ElementMultiAspect.classFullName, async (multiAspect) => {
+      if (batchedElementMultiAspects.length === 0) {
+        batchedElementMultiAspects.push(multiAspect);
+        return;
+      }
+
+      // element id changed so all element's aspects are in the array and can be exported
+      if (batchedElementMultiAspects[0].element.id !== multiAspect.element.id) {
+        this.handler.onExportElementMultiAspects(batchedElementMultiAspects);
+        await this.handler.trackProgress();
+        batchedElementMultiAspects = [];
+      }
+
+      batchedElementMultiAspects.push(multiAspect);
+    });
+
+    if (batchedElementMultiAspects.length > 0) {
+      // aspects that are left in the array have not been exported
+      this.handler.onExportElementMultiAspects(batchedElementMultiAspects);
+      await this.handler.trackProgress();
+    }
+  }
+
+  private async exportAspectsLoop<T extends ElementAspect>(baseAspectClass: string, exportAspect: (aspect: T) => Promise<void>) {
+    for await (const aspect of this.queryAspects<T>(baseAspectClass)) {
+      if (!this.shouldExportElementAspect(aspect)) {
+        continue;
+      }
+
+      await exportAspect(aspect);
+    }
+  }
+
+  private async *queryAspects<T extends ElementAspect>(baseElementAspectClassFullName: string) {
+    const aspectClassNameIdMap = new Map<string, Id64String>();
+
+    const getAspectClassesSql = `SELECT DISTINCT ECClassId as classId, (ec_classname(ECClassId)) as className FROM ${baseElementAspectClassFullName}`;
+    const aspectClassesQueryReader = this.sourceDb.createQueryReader(getAspectClassesSql);
```

Comment:
Yeah, I think we don't need to use primary connection here. Source shouldn't be changed here as it won't be updated since transformer only updates `targetDb` and even if `provenanceDb` would be `sourceDb`, provenance is not recorded for aspects.

Reply:
fair enough considering the new query